xarray/tests/test_dataset.py

```diff
@@ -2872,6 +2872,7 @@ def test_rename_old_name(self) -> None:
         # This shouldn't cause any problems.
         data.rename({"var1": "var2", "var2": "var1"})
 
+    @pytest.mark.filterwarnings("ignore:rename:UserWarning")
```

Comment:
An advantage of testing for the warning, rather than filtering it out, is that we are explicit about when we want the warning — we're less likely to be silencing overly noisy warnings, or failing to warn when we should be.